docs/guides/entry_points.md

```diff
@@ -0,0 +1,90 @@
+
+:::
+
+There are more cases where tests on your deliveries would need an Entry point. Let’s say you want to improve the box opening experience, include a note, etc. The assignment has to hapen when packaging; the entry point has to be set when the customer receives the note. There’s no reason to include customers who didn’t get their packages when measuring the impact of a hand-written note on their satisfaction.
+
+Not all logistical test require a separate Entry point. If you want to test a promotion that offers free delivery under certain condition, you can decide which customer sees that promotion when it becomes visible, not earlier.
+
+::: note
+
+Be mindful when testing promotions on anonymous users: you risk contaminating your experiment. If they get re-assigned by refreshing their browser, or opening a private window, some users will stumble on it, notice the discrepancy, and figure out the trick. They might leak that information. If prospective buyers see it, they will try. You are better off testing promotions on entities that buyers can’t change easily: established user account, phone number, postal address.
+
+:::
+
+### Third example: AI-generated customer service
+
+Finally, let’s say you want to test one of those chat-bots that promise to replave human-lead customer service. The vendor recommends to send them as much information about the customer as you can, including their queries and the product decription that they open during their session. Small issue: user logs are streamed to your data warehouse, but you don’t really store that information in the browser memory or a convenient session object that could be sent it as single blob to the AI company. No problem: the vendor can ingest page content as the visitor goes through your site, store them and will use those as context when a customer asks a question.
+
+Once again, you have to decide to send information about a customer, i.e. assign them to either Control or Treatment before they ask a question and are exposed to the AI bot. 
+
+In this example (as well as the examples about triggering complex recommendations) you could send an API call for **all users** and only decide which response to use once you need to show the information to the users. That pattern can avoid having to use Entry point. However, that vendor might charge per visitor, and you might want to lower the cost of testing their solution.
+
+More generally, complex models, some API calls, and AI services are non-trivial to scale and **expensive to serve**. You might want to limit the cost of testing Treatment by not wasting half of it. In a gradual roll-out process, you want organise your feature flag so that only internal developpers and testers see the results first, then expose a small minority of visitors to the A/B test, then scale it to a full 50/50 test. That allows you to control who you expose, and if Treatment is expensive, how much valuable ressources you dedicate to testing an idea.
```

Comment:
```suggestion
More generally, complex models, some API calls, and AI services are non-trivial to scale and **expensive to serve**. You might want to limit the cost of testing Treatment by not wasting half of it. In a gradual roll-out process, you want organise your feature flag so that only internal developers and testers see the results first, then expose a small minority of visitors to the A/B test, then scale it to a full 50/50 test. That allows you to control who you expose, and if Treatment is expensive, how much valuable resources you dedicate to testing an idea.
```